usaspending_api/etl/management/commands/elasticsearch_indexer.py

```diff
@@ -221,7 +223,63 @@ def parse_cli_args(options: dict, es_client) -> dict:
 
 
 def set_config(passthrough_values: list, arg_parse_options: dict) -> dict:
-    """Set values based on env vars and when the script started"""
+    """
+    Define the configurations for creating and populating each Elasticsearch index.
+
+        Config options:
+            base_table: Database table used to populate the Elasticsearch index.
+
+            base_table_id: The `id` column of the base_table.
+
+            create_award_type_aliases:
+
+            data_transform_func: Python function to call to format/convert/aggregate/drop the database data before it's
+                pushed to the Elasticsearch index.
+
+            data_type:
+
+            execute_sql_func: Function that will execute the appropriate SQL to retrieve data from the database to
+                populate the Elasticsearch index.
+
+                Default value: execute_sql_statement
+
+            extra_null_partition:
+
+            field_for_es_id: Field that will be used as the `id` value in the Elasticsearch index.
+
+            initial_datetime: Earliest date to return results for.
+                Default value: settings.API_SEARCH_MIN_DATE  (10/01/2007)
+
+            max_query_size: The maximum number of Elasticsearch results that can be retrieved in a single request.
+                Note: Setting this too high will hurt Elasticsearch performance.
+                Note: This should be defined in settings.py
+
+                Default value: 50000
+
+            optional_predicate: Optional SQl clause to add to the `sql_view` SQL statement.
+
+            primary_key:
+
+            query_alias_prefix: String used as a prefix for the Elasticsearch aliases.
+                Note: This should be defined in settings.py
+
+            required_index_name: Name of the Elasticsearch index.
+                Note: This should be defined in settings.py
+
+            sql_view: Name of the SQL View that will be used to populate the Elasticsearch index.
+                Note: This should be defined in settings.py
+
+            stored_date_key: The `name` in the `external_data_type` table to use when referencing this Elasticsearch
+                index. This table, in combination with the `external_data_load_date` table, are used to keep track
+                of when data was last loaded into a destination (database table or Elasticsearch index).
+
+            unique_key_field: Name of the Elasticsearch field that will be unique across all records.
+
+            write_alias: Name to use for the Elasticsearch alias that wil be used to write data.
+                Note: This should be defined in settings.py
+    """
```

Comment:
This documentation on the config was a nice addition. Thank you!

Reply:
Thanks! I hope the documentation makes it easier for others to update these configs, because I had to search through the code base to understand a few of these fields.